Stack/Opc.Ua.Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -250,31 +250,16 @@ public virtual void Validate(X509Certificate2Collection chain)
             catch (ServiceResultException se)
             {
                 // check for errors that may be suppressed.
-                switch (se.StatusCode)
+                if (containsUnsuppressibleSC(se.Result))
                 {
-                    case StatusCodes.BadCertificateHostNameInvalid:
-                    case StatusCodes.BadCertificateIssuerRevocationUnknown:
-                    case StatusCodes.BadCertificateChainIncomplete:
-                    case StatusCodes.BadCertificateIssuerTimeInvalid:
-                    case StatusCodes.BadCertificateIssuerUseNotAllowed:
-                    case StatusCodes.BadCertificateRevocationUnknown:
-                    case StatusCodes.BadCertificateTimeInvalid:
-                    case StatusCodes.BadCertificatePolicyCheckFailed:
-                    case StatusCodes.BadCertificateUseNotAllowed:
-                    case StatusCodes.BadCertificateUntrusted:
-                    {
-                        Utils.Trace("Certificate Vaildation failed for '{0}'. Reason={1}", certificate.Subject, (StatusCode)se.StatusCode);
-                        break;
-                    }
-
-                    default:
-                    {
-                        // write the invalid certificate to rejected store if specified.
-                        Utils.Trace((int)Utils.TraceMasks.Error, "Certificate '{0}' rejected. Reason={1}", certificate.Subject, (StatusCode)se.StatusCode);
-                        SaveCertificate(certificate);
+                    Utils.Trace((int)Utils.TraceMasks.Error, "Certificate '{0}' rejected. Reason={1}", certificate.Subject, (StatusCode)se.StatusCode);
+                    SaveCertificate(certificate);
 
-                        throw new ServiceResultException(se, StatusCodes.BadCertificateInvalid);
-                    }
+                    throw new ServiceResultException(se, StatusCodes.BadCertificateInvalid);
+                }
+                else
+                {
+                    Utils.Trace("Certificate Vaildation failed for '{0}'. Reason={1} (and possibly others)", certificate.Subject, (StatusCode)se.StatusCode);
```

Comment:
`se.ToLongString()` would give a detailed message with all inner status codes.
Better to use this instead of "_and possibly others_":

`Utils.Trace("Certificate Vaildation failed for '{0}'. Reason={1} ", certificate.Subject, se.ToLongString());`

Reply:
good point, but isn't .ToLongString also printing the call stack? That may be of less interest

Reply:
No, it should't print the call stack. 
I just made a test:

```
Certificate is not trusted.
SubjectName: CN=UaExpert@CW-AAM, O=Softing AG
IssuerName: CN=UaExpert@CW-AAM, O=Softing AG
Id: BadCertificateUntrusted
Description: Certificate is not trusted.
SubjectName: CN=UaExpert@CW-AAM, O=Softing AG
IssuerName: CN=UaExpert@CW-AAM, O=Softing AG
```